hal/src/thumbv7em/icm.rs

```diff
@@ -0,0 +1,1640 @@
+    #[inline]
+    pub fn disable_monitoring(&self) {
+        // Each bit in the register represents one of the
+        // four memory regions. Writing 0 does not change
+        // the state of the region, to enable the region,
+        // writing to `rmen` is required.
+        self.ctrl()
+            .write(|w| unsafe { w.rmdis().bits(self.mask()) });
+    }
+
+    /// Trigger recalculation of the memory monitor region
+    #[inline]
+    pub fn trigger_rehash(&self) {
+        self.ctrl()
+            .write(|w| unsafe { w.rehash().bits(self.mask()) });
+    }
+
+    /// Set Region Status Updated interrupt enable
+    #[inline]
+    pub fn set_rsu_int(&self) {
+        self.ier().write(|w| unsafe { w.rsu().bits(self.mask()) });
+    }
+
+    /// Set Region End bit Condition Detected interrupt enable
+    #[inline]
+    pub fn set_rec_int(&self) {
+        self.ier().write(|w| unsafe { w.rec().bits(self.mask()) });
+    }
+
+    /// Set Region Wrap Condition detected interrupt enable
+    #[inline]
+    pub fn set_rwc_int(&self) {
+        self.ier().write(|w| unsafe { w.rwc().bits(self.mask()) });
+    }
+
+    /// Set Region Bus Error interrupt enable
+    #[inline]
+    pub fn set_rbe_int(&self) {
+        self.ier().write(|w| unsafe { w.rbe().bits(self.mask()) });
+    }
+
+    /// Set Region Digest Mismatch interrupt enable
+    #[inline]
+    pub fn set_rdm_int(&self) {
+        self.ier().write(|w| unsafe { w.rdm().bits(self.mask()) });
+    }
+
+    /// Set Region Hash Completed interrupt enable
+    #[inline]
+    pub fn set_rhc_int(&self) {
+        self.ier().write(|w| unsafe { w.rhc().bits(self.mask()) });
+    }
+
+    /// Disable Region Status Updated interrupt enable
+    #[inline]
+    pub fn disable_rsu_int(&self) {
+        self.idr().write(|w| unsafe { w.rsu().bits(self.mask()) });
+    }
+
+    /// Disable Region End bit Condition Detected interrupt enable
+    #[inline]
+    pub fn disable_rec_int(&self) {
+        self.idr().write(|w| unsafe { w.rec().bits(self.mask()) });
+    }
+
+    /// Disable Region Wrap Condition detected interrupt enable
+    #[inline]
+    pub fn disable_rwc_int(&self) {
+        self.idr().write(|w| unsafe { w.rwc().bits(self.mask()) });
+    }
+
+    /// Disable Region Bus Error interrupt enable
+    #[inline]
+    pub fn disable_rbe_int(&self) {
+        self.idr().write(|w| unsafe { w.rbe().bits(self.mask()) });
+    }
+
+    /// Disable Region Digest Mismatch interrupt enable
+    #[inline]
+    pub fn disable_rdm_int(&self) {
+        self.idr().write(|w| unsafe { w.rdm().bits(self.mask()) });
+    }
+
+    /// Disable Region Hash Completed interrupt enable
+    #[inline]
+    pub fn disable_rhc_int(&self) {
+        self.idr().write(|w| unsafe { w.rhc().bits(self.mask()) });
+    }
+
+    /// Get Region Status Updated interrupt enable mask
+    #[inline]
+    pub fn get_rsu_int_mask(&self) -> bool {
+        matches!(self.imr().read().rsu().bits() & self.mask(), 1)
```

Comment:
You must have 30 copies of this pattern throughout the module. Would it be worth adding a function or macro for it?

Reply:
Good point, made it cleaner with some macros in https://github.com/atsamd-rs/atsamd/pull/480/commits/912991108dae800bc11dd5aadbff641a6e075ffc